crates/gitbutler-command-context/src/repository.rs

```diff
@@ -70,4 +71,51 @@ impl ProjectRepository {
     pub fn repo(&self) -> &git2::Repository {
         &self.git_repository
     }
+
+    /// Fetches a branches name without the remote name attached
+    ///
+    /// refs/heads/my-branch -> my-branch
+    /// refs/remotes/origin/my-branch -> my-branch
+    ///
+    /// An ideal implementation wouldn't require us to list all the references,
+    /// but there doesn't seem to be a libgit2 solution to this.
+    pub fn given_name_for_branch(&self, branch: &git2::Branch) -> Result<String> {
```

Comment:
I don't know how this is actually used, but if it is to obtain the remote name of a local tracking branch, then this wouldn't *always* work as refspecs need to be considered.
While `git2` doesn't have methods for this ([these probably lack what you are looking for](https://docs.rs/git2/latest/git2/struct.Repository.html#method.branch_remote_name)), `gix` has more but also doesn't provide a mapping from tracking branch to the ref name on the remote.

I have added this feature [to the `gitoxide` tracking issue for GitButler](https://github.com/Byron/gitoxide/issues/1287).

Reply:
We want to group items in the sidebar by the branch name, so we need to collect up all the remote branches with that given name, prs for branches with the given name, and local references with the given name.

GitHub's api gives you the `sourceBranch` property which is what I want, local references, by default don't have the remote name attached, so its just the remote branches that need this processing


<img width="305" alt="Screenshot 2024-07-12 at 14 30 31" src="https://github.com/user-attachments/assets/c488dea4-0725-4c29-b17b-ae37863388fb">